src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -46,22 +212,89 @@ class InvalidSchemaError(InvalidMessageError):
     pass
 
 
+class QuerySubscriptionStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(self, topic):
+        self.topic = topic
+
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        def process_message(message: Message[KafkaPayload]) -> None:
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=random() <= options.get("subscriptions-query.sample-rate"),
+            ), metrics.timer("snuba_query_subscriber.handle_message"):
+                try:
+                    handle_message(
+                        message.payload.value,
+                        message.value.offset,
+                        message.value.partition,
+                        self.topic,
+                    )
+                except Exception:
+                    # This is a failsafe to make sure that no individual message will block this
+                    # consumer. If we see errors occurring here they need to be investigated to
+                    # make sure that we're not dropping legitimate messages.
+                    logger.exception(
+                        "Unexpected error while handling message in QuerySubscriptionStrategy. Skipping message.",
+                        extra={
+                            "offset": message.value.offset,
+                            "partition": message.value.partition,
+                            "value": message.value,
+                        },
+                    )
+
+        return RunTask(process_message, CommitOffsets(commit))
+
+
+def get_query_subscription_consumer(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    strict_offset_reset: bool,
+    force_topic: Union[str, None],
+    initial_offset_reset: str = "earliest",
+) -> StreamProcessor[KafkaPayload]:
+    topic = force_topic or topic
+    cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+    cluster_options = kafka_config.get_kafka_consumer_cluster_options(
+        cluster_name,
+        {
+            "group.id": group_id,
+            "session.timeout.ms": 6000,
+            "auto.offset.reset": initial_offset_reset,
+            "enable.auto.commit": "false",
+            "enable.auto.offset.store": "false",
+            "enable.partition.eof": "false",
+            "default.topic.config": {"auto.offset.reset": initial_offset_reset},
+        },
```

Comment:
```suggestion
```

You should be able to remove these lines, the `build_kafka_consumer_configuration` function should build the config with the guarantees and configuration Arroyo expects.